ServerCore/Pages/Teams/Play.cshtml.cs

```diff
@@ -31,10 +31,15 @@ public PlayModel(PuzzleServerContext serverContext, UserManager<IdentityUser> us
 
         private const SortOrder DefaultSort = SortOrder.GroupAscending;
 
-        public async Task OnGetAsync(SortOrder? sort)
+        public async Task OnGetAsync(SortOrder? sort, int teamId)
         {
+            TeamID = teamId;
             Team myTeam = await UserEventHelper.GetTeamForPlayer(_context, Event, LoggedInUser);
-            if (myTeam != null)
+            if (teamId != myTeam.ID)
+            {
+                throw new Exception("Unauthorized");
```

Comment:
Should this return Unauthorized()? (you'd have to make the method return Task<IActionResult>

Reply:
Removed - since the page itself now has an auth check that covers this case